vnext/Microsoft.ReactNative/ReactHost/CrashManager.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "CrashManager.h"
+#include "ReactInstanceWin.h"
+
+#include <winrt/Windows.Storage.h>
+
+#include <WerApi.h>
+
+namespace Mso::React {
+
+// when calling SetUnhandledExceptionFilter the previous filter is returned
+// this should be saved so that if needed it can be called after the new unhandled exception has finished executing
+// this allows the Windows Error Reporting system to process the error and upload the data for processing
+LPTOP_LEVEL_EXCEPTION_FILTER g_previousExceptionFilter = nullptr;
+
+std::wstring g_logFileName;
+bool g_wasCalled{false}; // Prevent recursion in the custom handler
```

Comment:
>g_wasCalled [](http://example.com/codeflow?start=5&length=11)

(nit) since this is in general Mso::React namespace, can we make naming more distinct, eg exceptionFilterWasCalled, or something like that